crates/re_space_view_time_series/src/space_view_class.rs

```diff
@@ -0,0 +1,275 @@
+use egui::{
+    plot::{Legend, Line, Plot, Points},
+    Color32,
+};
+
+use re_arrow_store::TimeType;
+use re_format::next_grid_tick_magnitude_ns;
+use re_log_types::EntityPath;
+use re_space_view::{controls, EmptySpaceViewState};
+use re_viewer_context::{
+    SpaceViewClass, SpaceViewClassName, SpaceViewId, TypedScene, ViewerContext,
+};
+
+use crate::scene_part::{PlotSeriesKind, SceneTimeSeries};
+
+#[derive(Default)]
+pub struct TimeSeriesSpaceView;
+
+impl SpaceViewClass for TimeSeriesSpaceView {
+    type State = EmptySpaceViewState;
+    type Context = re_space_view::EmptySceneContext;
+    type SceneParts = SceneTimeSeries;
+    type ScenePartData = ();
+
+    fn name(&self) -> SpaceViewClassName {
+        "Time Series".into()
+    }
+
+    fn icon(&self) -> &'static re_ui::Icon {
+        &re_ui::icons::SPACE_VIEW_HISTOGRAM
+    }
+
+    fn help_text(&self, re_ui: &re_ui::ReUi, _state: &Self::State) -> egui::WidgetText {
+        let mut layout = re_ui::LayoutJobBuilder::new(re_ui);
+
+        layout.add("Pan by dragging, or scroll (+ ");
+        layout.add(controls::HORIZONTAL_SCROLL_MODIFIER);
+        layout.add(" for horizontal).\n");
+
+        layout.add("Zoom with pinch gesture or scroll + ");
+        layout.add(controls::ZOOM_SCROLL_MODIFIER);
+        layout.add(".\n");
+
+        layout.add("Drag ");
+        layout.add(controls::SELECTION_RECT_ZOOM_BUTTON);
+        layout.add(" to zoom in/out using a selection.\n");
+
+        layout.add("Click ");
+        layout.add(controls::MOVE_TIME_CURSOR_BUTTON);
+        layout.add(" to move the time cursor.\n\n");
+
+        layout.add_button_text(controls::RESET_VIEW_BUTTON_TEXT);
+        layout.add(" to reset the view.");
+
+        layout.layout_job.into()
+    }
+
+    fn preferred_tile_aspect_ratio(&self, _state: &Self::State) -> Option<f32> {
+        None
+    }
+
+    fn selection_ui(
+        &self,
+        _ctx: &mut ViewerContext<'_>,
+        _ui: &mut egui::Ui,
+        _state: &mut Self::State,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+    ) {
+    }
+
+    fn ui(
+        &self,
+        ctx: &mut ViewerContext<'_>,
+        ui: &mut egui::Ui,
+        _state: &mut Self::State,
+        scene: &mut TypedScene<Self>,
+        _space_origin: &EntityPath,
+        _space_view_id: SpaceViewId,
+    ) {
+        ui.scope(|ui| {
```

Comment:
Why add this `scope`? Could it be added at the call-site instead to avoid this rightward drift?

Reply:
took it over from before, but yeah I suppose all space views should live in a scope!